api/publish.go

```diff
@@ -415,3 +545,486 @@ func apiPublishDrop(c *gin.Context) {
+	if err != nil {
+		AbortWithJSONError(c, http.StatusNotFound, fmt.Errorf("unable to delete: %s", err))
+		return
+	}
+
+	err = collection.LoadComplete(published, collectionFactory)
+	if err != nil {
+		AbortWithJSONError(c, http.StatusInternalServerError, fmt.Errorf("unable to delete: %s", err))
+		return
+	}
+
+	revision := published.ObtainRevision()
+	sources := revision.Sources
+
+	_, exists := sources[component]
+	if !exists {
+		AbortWithJSONError(c, http.StatusNotFound, fmt.Errorf("unable to delete: Component '%s' does not exist", component))
+		return
+	}
+
+	delete(sources, component)
+
+	resources := []string{string(published.Key())}
+	taskName := fmt.Sprintf("Update published %s repository %s/%s", published.SourceKind, published.StoragePrefix(), published.Distribution)
+	maybeRunTaskInBackground(c, taskName, resources, func(out aptly.Progress, _ *task.Detail) (*task.ProcessReturnValue, error) {
+		err = collection.Update(published)
+		if err != nil {
+			return &task.ProcessReturnValue{Code: http.StatusInternalServerError, Value: nil}, fmt.Errorf("unable to save to DB: %s", err)
+		}
+
+		return &task.ProcessReturnValue{Code: http.StatusOK, Value: gin.H{}}, nil
+	})
+}
+
+type publishedRepoUpdateParams struct {
+	// when publishing, overwrite files in pool/ directory without notice
+	ForceOverwrite bool `                         json:"ForceOverwrite"  example:"false"`
+	// GPG options
+	Signing signingParams `                       json:"Signing"`
+	// Don't generate contents indexes
+	SkipContents *bool `                          json:"SkipContents"    example:"false"`
+	// Skip bz2 compression for index files
+	SkipBz2 *bool `                               json:"SkipBz2"         example:"false"`
+	// Don't remove unreferenced files in prefix/component
+	SkipCleanup *bool `                           json:"SkipCleanup"     example:"false"`
+	// Provide index files by hash
+	AcquireByHash *bool `                         json:"AcquireByHash"   example:"false"`
+	// Enable multiple packages with the same filename in different distributions
+	MultiDist *bool `                             json:"MultiDist"       example:"false"`
+}
+
+// @Summary Update Published Repository
+// @Description **Update a published repository**
+// @Description
+// @Description Apply pending changes and republish
+// @Description
+// @Description See also: `aptly publish update` / `aptly publish switch`
+// @Tags Publish
+// @Param prefix path string true "publishing prefix"
+// @Param distribution path string true "distribution name"
+// @Consume json
+// @Param request body publishedRepoUpdateParams true "Parameters"
+// @Produce json
+// @Success 200 {object} deb.PublishedRepo
+// @Failure 400 {object} Error "Bad Request"
+// @Failure 404 {object} Error "Published repository/component not found"
+// @Failure 500 {object} Error "Internal Error"
+// @Router /api/publish/{prefix}/{distribution}/update [post]
```

Comment:
should we use PUT here ? as the other 'update' apis are using PUT... POST would be only for creating the publish point...

Reply:
This is a special case here, because `/update` has more the semantics of an action call: https://github.com/NationalBankBelgium/REST-API-Design-Guide/wiki/REST-Resources-Actions

I normally use POST in those cases, because it is not a single resource that gets created and the action call may be triggered concurrently.